include/debug/gdbstub.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEBUG_GDBSTUB_H_
+#define ZEPHYR_INCLUDE_DEBUG_GDBSTUB_H_
+
+#include <arch/gdbstub.h>
+
+/* Map from CPU excpetions to GDB  */
+#define GDB_EXCEPTION_INVALID_INSTRUCTION   4UL
+#define GDB_EXCEPTION_BREAKPOINT            5UL
+#define GDB_EXCEPTION_MEMORY_FAULT          7UL
+#define GDB_EXCEPTION_DIVIDE_ERROR          8UL
+#define GDB_EXCEPTION_INVALID_MEMORY        11UL
+#define GDB_EXCEPTION_OVERFLOW              16UL
+
+/**
+ * @brief Start debugging
+ *
+ * This function will stop Zephyr execution and wait for a gdb
+ * connection.  Once gdb is connected it will loop waiting for a
+ * instruction.
+ *
+ * @note Just call this function when the selected backend was
+ * properly initialized. Be aware that different backends can have
+ * different initialization requirements.
+ */
+void gdb_start(void);
```

Comment:
who is the intended user of this function? the kernel's init.c? the application?

Reply:
I was not sure, right now I just remove this API and the debugging is initialized with `SYS_INIT` and no API is necessary.